p2panda-rs/src/schema/system_schema.rs

```diff
@@ -0,0 +1,327 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::convert::TryFrom;
+use std::str::FromStr;
+
+use crate::document::{DocumentView, DocumentViewId};
+use crate::operation::{OperationValue, Relation};
+
+use super::SystemSchemaError;
+
+#[derive(Clone, Debug, Copy, PartialEq)]
+#[allow(missing_docs)]
+pub enum FieldType {
+    Bool,
+    Int,
+    Float,
+    String,
+    Relation,
+}
+
+impl FromStr for FieldType {
+    type Err = SystemSchemaError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "bool" => Ok(FieldType::Bool),
+            "int" => Ok(FieldType::Int),
+            "float" => Ok(FieldType::Float),
+            "str" => Ok(FieldType::String),
+            "relation" => Ok(FieldType::Relation),
+            _ => Err(SystemSchemaError::InvalidFieldType),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SchemaView {
+    // ID of this schema view.
+    id: DocumentViewId,
+    /// Name of this schema.
+    name: String,
+    /// Description of this schema.
+    description: String,
+    /// The fields in this schema.
+    /// TODO: This will be a `relation` when we merge that PR
+    fields: RelationList,
```

Comment:
Very pendantic: when writing comments in struct fields we so far separated them with a newline in-between each field. I'd like to stick to it for nice style consistency ✨ (still need to do that style guide).

Same goes for `SchemaFieldView`.